sway-core/src/convert_parse_tree.rs

```diff
@@ -2552,14 +2552,12 @@ fn statement_let_to_ast_nodes(
         span: Span,
     ) -> Result<Vec<AstNode>, ErrorEmitted> {
         let ast_nodes = match pattern {
-            Pattern::Wildcard { .. } => {
-                let ast_node = AstNode {
-                    content: AstNodeContent::Expression(expression),
-                    span,
+            Pattern::Wildcard { .. } | Pattern::Var { .. } => {
+                let (mutable, name) = match pattern {
```

Comment:
I feel like there must be a better way to do this than to match on the same pattern twice, but nothing obvious springs to mind in this case.  Perhaps we can rethink how we manage wildcard patterns, and instead of having a separate variant for `Wildcard` just add an `is_wildcard` field to `Var`.

Dunno.  This is OK for now. :slightly_smiling_face: